src/fireedge/src/modules/components/Tabs/Cluster/Drs/index.js

```diff
@@ -0,0 +1,284 @@
+ * limitations under the License.                                            *
+ * ------------------------------------------------------------------------- */
+
+import { Tr } from '@modules/components/HOC'
+import { useEffect, ReactElement } from 'react'
+import PropTypes from 'prop-types'
+import { Box, Stack } from '@mui/material'
+import { partition, isEmpty } from 'lodash'
+
+import { ClusterAPI, useGeneralApi, useSystemData } from '@FeaturesModule'
+import ButtonToTriggerForm from '@modules/components/Forms/ButtonToTriggerForm'
+import { UpdatePlanConfigurationForm } from '@modules/components/Forms/Cluster'
+
+import { jsonToXml } from '@ModelsModule'
+import {
+  T,
+  DRS_CONFIG_ATTRIBUTES,
+  DRS_AUTOMATION,
+  STYLE_BUTTONS,
+} from '@ConstantsModule'
+import { sentenceCase } from '@UtilsModule'
+import { List } from '@modules/components/Tabs/Common'
+import ExecutionTimeline from '@modules/components/Tabs/Cluster/Drs/timeline'
+
+/**
+ * Renders configuration tab.
+ *
+ * @param {object} props - Props
+ * @param {string} props.id - Virtual machine id
+ * @returns {ReactElement} Configuration tab
+ */
+const PlanOptimization = ({ id }) => {
+  const { enqueueSuccess } = useGeneralApi()
+  const [updateConf] = ClusterAPI.useUpdateClusterMutation()
+  const [applyPlan] = ClusterAPI.useApplyPlanMutation()
+  const [deletePlan] = ClusterAPI.useDeletePlanMutation()
+  const { oneConfig: { DRS_INTERVAL = -1 } = {} } = useSystemData()
+
+  const [
+    fetchCluster,
+    { data: clusterData = {}, isFetching: fetchingCluster },
+  ] = ClusterAPI.useLazyGetClusterQuery()
+  const [
+    optimize,
+    { isFetching: fetchingOptimizationPlan, isSuccess: planGenerated },
+  ] = ClusterAPI.useOptimizeClusterMutation()
+
+  const { data: { TEMPLATE: { ONE_DRS = {} } } = {}, isFetching } =
+    ClusterAPI.useGetClusterQuery({
+      id,
+    })
+
+  const { PLAN: optimizationPlan = {} } = clusterData
+
+  const handleUpdateConf = async (newConfiguration) => {
+    const xml = jsonToXml(newConfiguration)
+    await updateConf({ id, template: xml, replace: 1 }) // Merge with existing
+  }
+
+  const handleApplyPlan = () => {
+    applyPlan({ id })
+  }
+
+  const handleDeletePlan = () => {
+    deletePlan({ id })
+  }
+
+  const handleDisableDrs = async () => {
+    const {
+      PLAN,
+      TEMPLATE: { ONE_DRS: _oneDrs, ...template },
+    } = clusterData
+    PLAN && handleDeletePlan()
+    const xml = jsonToXml(template)
+    await updateConf({ id, template: xml, replace: 0 }) // Merge with existing
+  }
+
+  useEffect(() => {
+    if (planGenerated) {
+      enqueueSuccess(T.OptimizationPlanGenerated)
+      fetchCluster({ id })
+    }
+  }, [planGenerated])
+
+  const handleOptimization = () => {
+    optimize({ id })
+  }
+
+  // Loads initial plan data if any
+  useEffect(() => {
+    fetchCluster({ id })
+  }, [])
+
+  const [weights, rest] = partition(
+    Object.values(DRS_CONFIG_ATTRIBUTES),
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).